hazelcast/src/test/java/com/hazelcast/query/impl/TypeConverterTest.java

```diff
@@ -39,4 +53,246 @@ Comparable convertInternal(Comparable value) {
+        ));
+    }
+
+    @Test
+    public void testBigIntegerConvert_whenPassedLongValue_thenConvertToBigInteger() throws Exception {
+        Long longValue = 3141593L;
+        Comparable expectedBigIntValue = BigInteger.valueOf(longValue.longValue());
+
+        Comparable comparable = TypeConverters.BIG_INTEGER_CONVERTER.convert(longValue);
+
+        assertThat(comparable, allOf(
+                is(instanceOf(BigInteger.class)),
+                is(equalTo(expectedBigIntValue))
+        ));
+    }
+
+    @Test
+    public void testBigIntegerConvert_whenPassedIntegerValue_thenConvertToBigInteger() throws Exception {
+        Integer integerValue = 3141593;
+        Comparable expectedBigIntValue = BigInteger.valueOf(integerValue.longValue());
+
+        Comparable comparable = TypeConverters.BIG_INTEGER_CONVERTER.convert(integerValue);
+
+        assertThat(comparable, allOf(
+                is(instanceOf(BigInteger.class)),
+                is(equalTo(expectedBigIntValue))
+        ));
+    }
+
+    @Test
+    public void testBigIntegerConvert_whenPassedBigDecimalValue_thenConvertToBigInteger() throws Exception {
+        BigDecimal value = BigDecimal.valueOf(4.9999);
+        Comparable expectedBigIntValue = BigInteger.valueOf(value.longValue());
+
+        Comparable comparable = TypeConverters.BIG_INTEGER_CONVERTER.convert(value);
+
+        assertThat(comparable, allOf(
+                is(instanceOf(BigInteger.class)),
+                is(equalTo(expectedBigIntValue))
+        ));
+    }
+
+    @Test
+    public void testBigIntegerConvert_whenPassedHugeBigDecimalValue_thenConvertToBigInteger() throws Exception {
+        BigDecimal value = BigDecimal.ONE.add(
+                BigDecimal.valueOf(Long.MAX_VALUE));
+        Comparable expectedBigIntValue = BigInteger.valueOf(Long.MAX_VALUE).add(BigInteger.ONE);
+
+        Comparable comparable = TypeConverters.BIG_INTEGER_CONVERTER.convert(value);
+
+        assertThat(comparable, allOf(
+                is(instanceOf(BigInteger.class)),
+                is(equalTo(expectedBigIntValue))
+        ));
+    }
+
+    @Test
+    public void testBigIntegerConvert_whenPassedBigIntegerValue_thenConvertToBigInteger() throws Exception {
+        BigInteger value = BigInteger.ONE;
+        Comparable expectedBigIntValue = BigInteger.valueOf(value.longValue());
+
+        Comparable comparable = TypeConverters.BIG_INTEGER_CONVERTER.convert(value);
+
+        assertThat(comparable, allOf(
+                is(instanceOf(BigInteger.class)),
+                is(equalTo(expectedBigIntValue))
+        ));
+    }
+
+    @Test
+    public void testBigIntegerConvert_whenPassedBooleanValue_thenConvertToBigInteger() throws Exception {
+        Boolean value = Boolean.TRUE;
+        Comparable expectedBigIntValue = BigInteger.ZERO;   // Boolean TRUE means 0
+                                                            // Boolean FALSE means something non-zero, i.e. 1
```

Comment:
It would also great to inline the trailing comments. In general that's in the checkstyle rules but they are not evaluated against the test code for now.